### ISSUE TITLE
Tweak to search keyword inside of brackets?

### ISSUE THREAD

**Issue by jacobgraf** [blainesch/alfred-chrome-bookmarks#19]
First off, thanks for making this, it works great by default, but I am looking to tweak it to modify the search query to be inside of brackets []. So if I search 'email', I really want your script to search my Chrome bookmarks for '[email]'. I looked through your bookmarks.php file and added this to the top of the BookmarkModel find method but it doesn't seem to be working.

`$term = "[".$term."]";`

My guess is that some of your regex code is pruning out my brackets? I am by no means a regex expert and thought I would post this here to see if maybe you knew what was causing it not to work? Thanks for your time!

**Comment by blainesch** [comment]
@jacobgraf I'm glad you're enjoying the plugin!

The regular expression it generates for `[email]` is really complicated. It will in fact try and match `[email]` directly, but it allows for typos, so the regex is a lot more complex. For instance when you type `[email]` it will still match `email` by itself.

Here is a [regex 101](https://regex101.com/r/fY8mG0/1) link showing how it'll match the term correctly. There is even [a test](https://github.com/blainesch/alfred-chrome-bookmarks/blob/master/tests/src/QueryTest.php#L19-L21) to ensure brackets match correctly.

Can you give me more detail on why you don't think it's matching correctly?

**Comment by jacobgraf** [comment]
Thanks for the quick reply. Essentially I am trying to mimic the functionality of this Chrome extension. (https://chrome.google.com/webstore/detail/quickmarks/piefpokhpcehbeelhohgcnbipnfkogig?hl=en)

I really don't need any advanced searching, basically, I want to define 'keywords' in my Chrome bookmark titles inside of brackets. They will all be single word, no spaces. I don't want Alfred to look at anything except what's in those brackets so, as long as I only have one bookmark per keyword, I can quickly type the keyword and hit enter.

From the looks of it, your plugin is much more complicated than I need so maybe I can try to dumb it down to act like I've outlined above. Is that something easy to do?

**Comment by blainesch** [comment]
@jacobgraf there is a way to do this, but this plugin should still rank `[email]` over `email`. Do you have a few bookmarks you can share where it either doesn't match or rank as you might expect?

**Comment by jacobgraf** [comment]
But I think you're missing my point. I don't want to type '[email]' in as a query, I want to simply type 'email' and if it doesn't find a bookmark with '[email]' in the title, it won't return any results. Does that make sense?

**Comment by blainesch** [comment]
The package on packal was out dated. I uploaded the new v1.1 vesrion.

If you install this package you can follow these steps:

1) Click on the plugin, then double click the script filter:
<img width="1102" alt="alfred_preferences_and_tweak_to_search_keyword_inside_of_brackets__ _issue__19_ _blainesch_alfred-chrome-bookmarks" src="https://cloud.githubusercontent.com/assets/769039/16750007/43a3b29a-4783-11e6-9078-225eb47059b8.png">

2) Change `{query}` to `[{query}]`
<img width="1103" alt="alfred_preferences_and_tweak_to_search_keyword_inside_of_brackets__ _issue__19_ _blainesch_alfred-chrome-bookmarks" src="https://cloud.githubusercontent.com/assets/769039/16750030/63971f92-4783-11e6-84bb-634a29c7aa6f.png">

This should start searching correctly. I added two bookmarks `[Zwitter]` and `Zwitter` and after making this change `[Zwitter]` appears first when I search for just `Zwitter`.

Let me know if this works or not.

**Comment by jacobgraf** [comment]
That works better! Ideally, I'd like it to not show any results except based on what's in those brackets, but I can't figure out how to dumb it down to operate that way, so this will work for now. Thanks!
